homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,326 @@
+
+
+def _get_id_from_name(name, dictionary):
+    for key in dictionary.keys():
+        # Replace unicode non-breaking space (used in Nobø Ecohub) with space
+        if dictionary[key][ATTR_NAME].replace("\xa0", " ") == name:
+            return key
+    return None
+
+
+class NoboZone(ClimateEntity):
+    """Representation of a Nobø zone.
+
+    A Nobø zone consists of a group of physical devices that are
+    controlled as a unity.
+    """
+
+    _attr_max_temp = MAX_TEMPERATURE
+    _attr_min_temp = MIN_TEMPERATURE
+    _attr_precision = PRECISION_TENTHS
+    _attr_preset_modes = PRESET_MODES
+    # Need to poll to get preset change when in HVAC_MODE_AUTO.
+    _attr_should_poll = True
+    _attr_supported_features = SUPPORT_FLAGS
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(
+        self, zone_id, hub: nobo, command_off_id, command_on_id, override_type
+    ):
+        """Initialize the climate device."""
+        self._id = zone_id
+        self._nobo = hub
+        self._attr_unique_id = hub.hub_serial + ":" + zone_id
+        self._attr_name = hub.zones[self._id][ATTR_NAME]
+        self._attr_hvac_mode = HVAC_MODE_AUTO
+        self._command_off_id = command_off_id
+        self._command_on_id = command_on_id
+        self._override_type = override_type
+        self._can_turn_off = (
+            self._command_on_id is not None and self._command_off_id is not None
+        )
+
+        # Register for callbacks before initial update to avoid race condition.
+        hub.register_callback(self._after_update)
+        self.update()
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        # Only enable off-command if on- and off-command exists for this zone:
+        if self._can_turn_off:
+            return HVAC_MODES
+        return HVAC_MODES_WITHOUT_OFF
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set HVAC mode to comfort(HEAT) or back to normal(AUTO)."""
+        if hvac_mode == HVAC_MODE_AUTO:
+            await self.async_set_preset_mode(PRESET_NONE)
+            self._attr_hvac_mode = hvac_mode
+        elif hvac_mode == HVAC_MODE_HEAT:
+            await self.async_set_preset_mode(PRESET_COMFORT)
+            self._attr_hvac_mode = hvac_mode
+
+        if self._can_turn_off:
+            if hvac_mode == HVAC_MODE_OFF:
+                await self.async_set_preset_mode(PRESET_NONE)
+                self._attr_hvac_mode = hvac_mode
+                await self._nobo.async_update_zone(
+                    self._id, week_profile_id=self._command_off_id
+                )  # Change week profile to OFF
+                _LOGGER.debug(
+                    "Turned off heater %s '%s' by switching to week profile %s",
+                    self._id,
+                    self._attr_name,
+                    self._command_off_id,
+                )
+            else:
+                await self._nobo.async_update_zone(
```

Comment:
Why does this only run when `_can_turn_off` is true? It is not obvious reading this why this happens since naively it looks like it supposed to run anytime the hvac mode isn't off.

Reply:
`HVAC_MODE_OFF` (if supported) changes the week profile to the "off" profile. On `HVAC_MODE_HEAT` or `HVAC_MODE_AUTO`, we must return to the normal week profile. I think this can be cleaned up to check current HVAC mode first, and only set week profile if necessary.

Reply:
Fixed.